### PR TITLE
[OTTO-395] set WP_ENVIRONMENT_TYPE according to Pantheon environment

### PR DESCRIPTION
**Pull request by pwtyler**
WordPress 5.5 introduces `wp_get_environment_type()`, which returns a site's environment type as defined by the constant `WP_ENVIRONMENT_TYPE`. 

The `live` environment returns `production`, `test` returns `staging`, and all other environments (`dev`, all multidevs) return `development`.

References:
https://make.wordpress.org/core/2020/07/24/new-wp_get_environment_type-function-in-wordpress-5-5/

### CODE REVIEW

File: wp-config-pantheon.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
+
```
Comment (pwtyler) [inline wp-config-pantheon.php:2]: TODO: Add Comment: `DO NOT EDIT`

**Comment by westonruter** [comment]
Would it not rather be preferable to specify this not in the `wp-config.php` but rather in environment variables, similar to how `PANTHEON_ENVIRONMENT` is defined? WordPress first checks `getenv( 'WP_ENVIRONMENT_TYPE' )` before checking for a `WP_ENVIRONMENT_TYPE` constant:

https://github.com/WordPress/wordpress-develop/blob/1c6ca0998c02a5c569a526324bbe2a22eb1ee9a6/src/wp-includes/load.php#L170-L181

This will reduce merge conflict conflicts with `wp-config.php`.

File: wp-config.php [hunk 2]
```diff
@@ -88,7 +88,11 @@
     if ( in_array( $_ENV['PANTHEON_ENVIRONMENT'], array( 'test', 'live' ) ) && ! defined( 'DISALLOW_FILE_MODS' ) ) :
         define( 'DISALLOW_FILE_MODS', true );
     endif;
-
+    
+    if (file_exists(dirname(__FILE__) . '/wp-config-pantheon.php')) :
```
Comment (greg-1-anderson) [inline wp-config.php:92]: As long as we are touching wp-config.php, let's move all of the Panthon-specific changes to the wp-config-panthon.php file. That way, if we have future need to change any Pantheon-specific behavior, we can do it without touching wp-config.php.

**Comment by greg-1-anderson** [comment]
Thanks for the feedback, @westonruter. We did actually consider that strategy; however, it is more effort to do it that way, so making the change in the WordPress upstream had a number of advanteges:

1. We could guarantee that the update can be deployed before the new WordPress release with the feature comes out without sacrificing other projects.

2. Having the change in the WordPress upstream directly advertises that the feature is supported in Pantheon.

3. Making the change in WordPress now might cause merge conflicts for some people, but if we also use this opportunity to include a Pantheon-specific WordPress include file, as we should have done in the first place (and as we have done in the Drupal upstreams), then the next time we are faced with this decision, we can modify wp-config-pantheon.php and not cause merge conflicts.

File: wp-config-pantheon.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ * Set WP_ENVIRONMENT_TYPE according to the Pantheon Environment
+ */
+if (! defined('WP_ENVIRONMENT_TYPE')) {
```
Comment (greg-1-anderson) [inline wp-config-pantheon.php:6]: We should probably use getenv / putenv here.

If in the future we decided to move this logic to the prepend file, we could define WP_ENVIRONMENT in the prepend file, and that would effectively disable the code here, as intended. However, if we later decided to move this logic into environment variables, then the code here would take precedence over what we did in the platform.

While it's likely that the platform logic would be the same as what is here, we don't know if there might be some future need to do something different. So, it's a pretty minor point, but I think that putenv would be a little more flexible than define.

File: wp-config.php (hunk 2 above)
Comment (dylanjameswagner) [reply wp-config.php:92]: Is this introducing a new file named `wp-config-panthon.php`? Doesn't that open the door for moving all non-WordPress Core wp-config modifications to that file? Probably everything under the comment `/** A couple extra tweaks to help things run well on Pantheon. **/`.

Comment (greg-1-anderson) [reply wp-config.php:92]: Yes, that was my suggestion exactly.

File: wp-config-pantheon.php (hunk 3 above)
Comment (pwtyler) [reply wp-config-pantheon.php:6]: Overriding an ENV variable is also an option for more bespoke workflows where re-setting a constant is not, so another reason for putenv 👍

**Comment by pwtyler** [comment]
Moved remainder of Pantheon-specific block into `wp-config-pantheon.php` and moved WP_ENVIRONMENT_TYPE from constants to putenv(). Added incidental change from #245 per request there. Will squash/fixup before merge once this passes further CR.

File: .gitignore [hunk 4]
```diff
@@ -13,7 +13,6 @@ wp-content/advanced-cache.php
 wp-content/wp-cache-config.php
 sitemap.xml
 sitemap.xml.gz
-*.log
```
Comment (greg-1-anderson) [inline .gitignore:16]: Why was this removed? I'm neutral on this change, just wondering.

File: wp-config-pantheon.php [hunk 5]
```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Pantheon platform settings. Everything you need should already be set.
+ * To avoid merge conflicts, no changes should be introduced to this file
```
Comment (greg-1-anderson) [inline wp-config-pantheon.php:4]: Don't say "to avoid merge conflicts", just say that editing is not allowed.

On the Drupal side, the comment is:
```
/**
 * @file
 * Pantheon configuration file.
 *
 * IMPORTANT NOTE:
 * Do not modify this file. This file is maintained by Pantheon.
 *
 * Site-specific modifications belong in settings.php, not this file. This file
 * may change in future releases and modifications would cause conflicts when
 * attempting to apply upstream updates.
 */
```

File: wp-config.php [hunk 6]
```diff
@@ -24,71 +24,9 @@
  */
 else:
   if (isset($_ENV['PANTHEON_ENVIRONMENT'])):
```
Comment (greg-1-anderson) [inline wp-config.php:26]: As long as we're modifying this section, let's switch from the odd `if ():` syntax, and use `{}` as usual here.

File: .gitignore (hunk 4 above)
Comment (pwtyler) [reply .gitignore:16]: duplicated below (line 42). Originally opened in https://github.com/pantheon-systems/WordPress/pull/219, just been sitting for a while.

File: wp-config-pantheon.php (hunk 5 above)
Comment (pwtyler) [reply wp-config-pantheon.php:4]: That was definitely a sentence I meant to circle back to (I accidentally the `.` too). Thanks.

File: wp-config.php (hunk 6 above)
Comment (pwtyler) [reply wp-config.php:26]: Already tried that— mixing the two `if` syntaxes did not play nice. We can replace *all* the if / endif throughout the file to use {} if we want, or stay consistent with if/endif— with the increased chance of conflicts (which may be a moot point given we're already expecting some).
